core/trino-main/src/main/java/io/trino/execution/scheduler/EventDrivenFaultTolerantQueryScheduler.java

```diff
@@ -1150,14 +1153,16 @@ private void createStageExecution(SubPlan subPlan, boolean rootFragment, Map<Sta
 
                 boolean coordinatorStage = stage.getFragment().getPartitioning().equals(COORDINATOR_DISTRIBUTION);
 
+                boolean noMemoryFragment = isNoMemoryFragment(fragment);
                 StageExecution execution = new StageExecution(
                         queryStateMachine,
                         taskDescriptorStorage,
                         stage,
                         taskSource,
                         sinkPartitioningScheme,
                         exchange,
-                        memoryEstimatorFactory.createPartitionMemoryEstimator(),
+                        noMemoryFragment,
+                        noMemoryFragment ? new NoMemoryPartitionMemoryEstimator() : memoryEstimatorFactory.createPartitionMemoryEstimator(),
```

Comment:
can the standard estimator be modified to more adequatly  estimate memory usage of information_schema queries?
like
- if source stage and this is information_schema assume 0 memory need and ~0 (negligible) number of rows produced
- for intermediate stage, and sources produce ~0 rows, assume 0 memory need

(actually we should get rid of intermediate stages for DESCRIBE, it's nonsense)

Reply:
Having smarter memory estimation in general is a planned item - but much more complex than this fix. And will require some of a restructuring of interfaces. I think this is fine to have this as a stopgap for sake of metadata queries which are a very common outlier even for clusters which tend to run massing ETL workloads. 

Or maybe I did not understand your suggestion here. Let's chat tomorrow.